### PR TITLE
Add support for relative urls in webview

### LINKED ISSUE #907: HTML file wont load in android app
I have a HTML link that loads fine using my sitemaps when loading it using google chrome.

But in the app, this HTML link does not load

Its a pre configured widget from darksky weather. I have an example.html file in /var/lib in the webapps and weather layouts. That all works well and updates when i change the file.

But nothing to do with the example.html file is loading using the android app. Is this is a limitation of the android operating system, the app, or what?

my html file literally has one line 
<script type='text/javascript' src='https://darksky.net/widget/default/-33.771000,150.906300/ca12/en.js?width=100%&height=350&title=Blacktown&textColor=ffffff&bgColor=transparent&transparency=true&skyColor=bfbfbf&fontFamily=Default&customFont=&units=ca&htColor=cccccc&ltColor=636363&displaySum=yes&displayHeader=yes'></script>

### PR DESCRIPTION
**Pull request by mueller-ma**
Fixes #907

Signed-off-by: mueller-ma <mueller-ma@users.noreply.github.com>

### CODE REVIEW

File: mobile/src/main/java/org/openhab/habdroid/ui/OpenHABWidgetAdapter.java [hunk 1]
```diff
@@ -1001,11 +1003,16 @@ public void bind(OpenHABWidget widget) {
                 mWebView.setLayoutParams(lp);
             }
 
-            mWebView.setWebViewClient(new AnchorWebViewClient(widget.url(),
+            HttpUrl url = HttpUrl.parse(widget.url());
+            if (url == null && mConnection.getOpenHABUrl() != null) {
+                url = HttpUrl.parse(mConnection.getOpenHABUrl().toString() + widget.url());
```
[the remark this answers is not in the document]
Comment (maniac103) [reply mobile/src/main/java/org/openhab/habdroid/ui/OpenHABWidgetAdapter.java]: I'd suggest using HttpClient.buildUrl(). If it doesn't correctly handle this situation yet, we should make that method support it.

[the remark this answers is not in the document]
Comment (maniac103) [reply mobile/src/main/java/org/openhab/habdroid/ui/OpenHABWidgetAdapter.java]: Thinking about it, the mentioned method essentially does what your code does as well. Stripping the leading slash should be no problem though, and it's better to centralize that logic anyway.

[the remark this answers is not in the document]
Comment (mueller-ma) [reply mobile/src/main/java/org/openhab/habdroid/ui/OpenHABWidgetAdapter.java]: What's the best way to access `HttpClient.buildUrl()`? By making it static?

File: mobile/src/main/java/org/openhab/habdroid/util/HttpClient.java [hunk 2]
```diff
@@ -119,6 +119,9 @@ public void removeHeader(String key) {
     public HttpUrl buildUrl(String url) {
         HttpUrl absoluteUrl = HttpUrl.parse(url);
         if (absoluteUrl == null && mBaseUrl != null) {
+            if (url.indexOf("/") == 0 && url.length() > 1) {
```
Comment (maniac103) [inline mobile/src/main/java/org/openhab/habdroid/util/HttpClient.java]: Can simplify this to `if (url.startsWith("/")) {`

**Pull request merged**